src/templates/dandelion/config/screens/LockScreen.js

```diff
@@ -0,0 +1,338 @@
+    setFormError(null)
+    updateField(['lockToken', { data: token, selectedIndex }])
+  }, [])
+
+  const handleSpamPenaltyChange = useCallback(event => {
+    setFormError(null)
+    updateField(['spamPenalty', event.target.value])
+  }, [])
+
+  const handleLockAmountChange = useCallback(event => {
+    const value = parseInt(event.target.value, 10)
+    setFormError(null)
+    updateField(['lockAmount', isNaN(value) ? -1 : value])
+  }, [])
+
+  const prevNextRef = useRef()
+
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+      const error = validationError(
+        lockDuration,
+        lockToken.data.address,
+        lockAmount
+      )
+      setFormError(error)
+
+      if (!error) {
+        const screenData = {
+          lockToken: lockToken,
+          lockDuration,
+          spamPenalty,
+          lockAmount,
+        }
+        next(dataKey ? { ...data, [dataKey]: screenData } : screenData)
+      }
+    },
+    [lockDuration, lockToken, lockAmount, spamPenalty, next, dataKey, data]
+  )
+
+  return (
+    <form
+      css={`
+        display: grid;
+        align-items: center;
+        justify-content: center;
+      `}
+    >
+      <Header
+        title={title}
+        subtitle={
+          <span
+            css={`
+              display: flex;
+              align-items: center;
+              justify-content: center;
+            `}
+          >
+            Choose your
+            <span
+              css={`
+                display: flex;
+                margin: 0 ${1.5 * GU}px;
+              `}
+            >
+              <KnownAppBadge
+                appName="time-lock.aragonpm.eth"
+                label={appLabel}
+              />
+            </span>
+            settings below.
+          </span>
+        }
+      />
+
+      <div
+        css={`
+          ${fieldsLayout};
+        `}
+      >
+        <Field
+          label={
+            <React.Fragment>
+              Token address
+              <Help hint="What’s the token address?">
+                <strong>Token address</strong> is the address of the token to be
+                locked.
+              </Help>
+            </React.Fragment>
+          }
+        >
+          <TokenSelector
+            selectedIndex={lockToken.selectedIndex}
+            onChange={handleTokenChange}
+            value={lockToken.data.address}
+            tokens={[DEFAULT_LOCK_TOKEN]}
+          />
+        </Field>
+
+        <Field
+          label={
+            <React.Fragment>
+              Lock amount
+              <Help hint="What’s the lock amount?">
+                <strong>Lock amount</strong> is the amount of tokens to lock.
+              </Help>
+            </React.Fragment>
+          }
+        >
+          <TextInput
+            onChange={handleLockAmountChange}
+            value={lockAmount === -1 ? '' : lockAmount}
+            wide
```

Comment:
Probably doesn't need to be wide; it looks kind of weird

Reply:
Not sure what you mean. Do you see it different from this?

<img width="649" alt="Screen Shot 2020-02-13 at 8 06 25 PM" src="https://user-images.githubusercontent.com/22663232/74486721-5cfa6580-4e9c-11ea-81e2-4b72fc4eb80b.png">

If we don't set the `wide`, it sets width as `auto` and overflows the `div` container